components/date-picker/date-picker.component.ts

```diff
@@ -421,4 +422,19 @@ export class NzDatePickerComponent implements OnInit, OnChanges, OnDestroy, Cont
       }
     }
   }
+
+  onKeydown(event: KeyboardEvent): void {
+    if (!this.isRange && event.key === 'Tab') {
+      this.close();
+    } else if (this.isRange && event.key === 'Tab') {
+      if (this.datePickerService.activeInput === 'left' && event.shiftKey) {
```

Comment:
The `tab` action may be changed in the future. I think this way is better: 
```ts
if (!this.elementRef.nativeElement.contains(document.activeElement)) { 
  // ...
}
```

Reply:
I had to check on focusEvent since at both keyDown and keyUp, the nativeElement and activeElement would be the same.